src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.cs

```diff
@@ -103,7 +103,8 @@ public override void Initialize(AnalysisContext context)
                     }
 
                     // Must be literal array
-                    if (arrayCreationOperation.Initializer.ElementValues.Any(x => x is not ILiteralOperation))
+                    if (arrayCreationOperation.Initializer != null &&
+                        arrayCreationOperation.Initializer.ElementValues.Any(x => x is not ILiteralOperation))
```

Comment:
Only stylistic suggestion to avoid repeating `arrayCreationOperation.Initializer`

```suggestion
                    if (arrayCreationOperation.Initializer is { } initializer &&
                        initializer.ElementValues.Any(x => x is not ILiteralOperation))
```